### PR TITLE
motor_control_node Improvements

### PR DESCRIPTION
**Pull request by Isopod00**
We haven't figured out position control or current control yet, but we have some good changes we'd like to merge.

### CODE REVIEW

File: src/motor_control/src/motor_control_node.cpp [hunk 1]
```diff
@@ -58,119 +55,141 @@ class MotorControlNode : public rclcpp::Node {
...
+    // RCLCPP_INFO(this->get_logger(), "Setting the current of CAN ID: %u to %f amps", id, current); // Print Statement
   }
 
   // Get the current duty cycle of the motor
   float vesc_get_duty_cycle(uint32_t id) {
     if (std::chrono::steady_clock::now() - this->can_data[id].timestamp < this->threshold) {
       return this->can_data[id].dutyCycle;
     } else {
-      return -1; // Return -1 if the data is stale
+      return -999; // Return -999 if the data is stale
```
Comment (jmblixt3) [inline src/motor_control/src/motor_control_node.cpp]: Is using a sentinel value the best solution here?

File: src/motor_control/src/motor_control_node.cpp [hunk 2]
```diff
@@ -58,119 +55,141 @@ class MotorControlNode : public rclcpp::Node {
...
         "motor/get", std::bind(&MotorControlNode::get_callback, this, _1, _2));
 
-    // Initialize timers here
+    // Initialize timers below //
     timer = this->create_wall_timer(500ms, std::bind(&MotorControlNode::timer_callback, this));
 
     // Initialize publishers and subscribers below //
     // The name of this topic is determined by ros2socketcan_bridge
-    can_pub = this->create_publisher<can_msgs::msg::Frame>("CAN/slcan0/transmit", 100);
+    can_pub = this->create_publisher<can_msgs::msg::Frame>("CAN/can1/transmit", 100);
```
Comment (jmblixt3) [inline src/motor_control/src/motor_control_node.cpp]: Move this to arg/config file

File: src/motor_control/src/motor_control_node.cpp [hunk 3]
```diff
@@ -58,119 +55,141 @@ class MotorControlNode : public rclcpp::Node {
...
     // The name of this topic is determined by ros2socketcan_bridge
-    can_sub = this->create_subscription<can_msgs::msg::Frame>("CAN/slcan0/receive", 10,
-                                                              std::bind(&MotorControlNode::CAN_callback, this, _1));
+    can_sub = this->create_subscription<can_msgs::msg::Frame>("CAN/can1/receive", 10, std::bind(&MotorControlNode::CAN_callback, this, _1));
```
Comment (jmblixt3) [inline src/motor_control/src/motor_control_node.cpp]: Move this to arg/config file

File: src/motor_control/src/motor_control_node.cpp [hunk 4]
```diff
@@ -207,16 +226,16 @@ class MotorControlNode : public rclcpp::Node {
                     std::shared_ptr<rovr_interfaces::srv::MotorCommandGet::Response> response) {
     if (request->type == "velocity") {
       response->result = vesc_get_velocity(request->can_id);
-      response->success = response->result == -1 ? 0 : 1;
+      response->success = response->result == -999 ? 0 : 1;
```
Comment (jmblixt3) [inline src/motor_control/src/motor_control_node.cpp]: Same comment about sentinel values

**Review by jmblixt3 — CHANGES_REQUESTED** [review]
Determine a better solution than sentinel values and move potential different values to args/config file
Consider std::optional instead of sentinel values

File: src/motor_control/src/motor_control_node.cpp (hunk 3 above)
Comment (Isopod00) [reply src/motor_control/src/motor_control_node.cpp]: Done

File: src/motor_control/src/motor_control_node.cpp (hunk 2 above)
Comment (Isopod00) [reply src/motor_control/src/motor_control_node.cpp]: Done

File: src/motor_control/src/motor_control_node.cpp (hunk 1 above)
Comment (Isopod00) [reply src/motor_control/src/motor_control_node.cpp]: fixed using std::optional

File: src/motor_control/src/motor_control_node.cpp (hunk 4 above)
Comment (Isopod00) [reply src/motor_control/src/motor_control_node.cpp]: fixed using std::optional

**Review by jmblixt3 — APPROVED** [review]
Looks good

**Pull request merged**